src/librbd/ImageCtx.cc

```diff
@@ -59,7 +59,7 @@ class ThreadPoolSingleton : public ThreadPool {
                  "rbd_op_threads") {
     start();
   }
-  virtual ~ThreadPoolSingleton() {
+  ~ThreadPoolSingleton() {
```

Comment:
So, is `~destructor() override` OK? I could make it automatically.

btw, @jcfischer @cbodley take a look please.